agent/util-scripts/pbench-tool-meister-start

```diff
@@ -357,109 +994,19 @@ def main(_prog, cli_params):
-            # The Tool Meister host name is not the same as "localhost" so we
-            # can add it to the list of host names the Redis server will use.
-            hostnames_l.append(tm_bind_hostname)
-        else:
-            assert (
-                "localhost" in hostnames_l
-            ), f"Logic Bomb! localhost does not map to an IP"
-            # Whatever the tm_bind_hostname was it maps to the same IP address as
-            # localhost, so just use "localhost" for the Tool Meister host
-            # name.
-            tm_bind_hostname = "localhost"
-    hostnames = " ".join(hostnames_l)
-
     # +
-    # Step 2. - Start the Redis Server
+    # Step 2. - Start the Redis Server (optional)
     # -
-    if cli_params.redis_server is None:
-        # Create the Redis server pbench-specific configuration file
-        redis_conf = tm_dir / "redis.conf"
-        params = {
-            "hostnames": hostnames,
-            "tm_dir": tm_dir,
-            "redis_port": def_redis_port,
-        }
-        try:
-            with redis_conf.open("w") as fp:
-                fp.write(redis_conf_tmpl.format(**params))
-        except Exception:
-            logger.exception("failed to create redis server configuration")
-            return ReturnCode.EXCREDISCONFIG
 
-        # Start the Redis Server itself
-        redis_srvr = "redis-server"
-        redis_srvr_path = find_executable(redis_srvr)
-        redis_pid = tm_dir / f"redis_{def_redis_port:d}.pid"
+    if redis_server.managed():
         logger.debug("2. starting redis server")
+        redis_server.determine_ip_and_port(full_hostname)
         try:
-            retcode = os.spawnl(os.P_WAIT, redis_srvr_path, redis_srvr, redis_conf)
-        except Exception:
-            logger.exception("failed to create redis server, daemonized")
-            return ReturnCode.EXCSPAWNREDIS
-        else:
-            if retcode != 0:
-                logger.error(
-                    "failed to create redis server, daemonized; return code: %d",
-                    retcode,
-                )
-                return ReturnCode.REDISFAILED
-        redis_host = "localhost"
-        redis_port = def_redis_port
-    else:
-        parts = cli_params.redis_server.split(":", 1)
-        if len(parts) != 2:
-            logger.error("Bad Redis server specified, '%s'", cli_params.redis_server)
-            return ReturnCode.BADREDISARG
-        try:
-            redis_port = int(parts[1])
-        except ValueError:
-            logger.error("Bad Redis port specified, '%s'", cli_params.redis_server)
-            return ReturnCode.BADREDISPORT
-        else:
-            redis_host = parts[0]
+            redis_server.start(tm_dir, full_hostname, logger)
+        except redis_server.Err as exc:
+            logger.error(f"Failed to start a local Redis server: '{exc}'")
+            # return ReturnCode.EXCREDISCREATION
+            return exc.return_code
```

Comment:
Why is there commented-out code, here?  I recommend either implementing it or removing it (or adding an appropriately explanatory comment with a `FIXME` or `TODO`).